api/api.go

```diff
@@ -458,9 +460,104 @@ func setExtensionUrls(original core.Extension, rootUrl string) core.Extension {
 		}
 	}
 
+	localization, err := GetLocalization(&extension)
+	if err != nil {
+		log.Printf("failed to retrieve locales: %v", err)
+	}
+
+	extension.Localization = localization
+
 	return extension
 }
 
+func GetLocalization(extension *core.Extension) (*core.Localization, error) {
+	path := filepath.Join(".", extension.Development.RootDir, "locales")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// The extension does not have a locales directory.
+		return nil, nil
+	}
+
+	fileNames, err := GetFileNames(path)
+	if err != nil {
+		return nil, err
+	}
+	translations := make(map[string]interface{})
+	defaultLocale := ""
+
+	for _, fileName := range fileNames {
+		data, err := GetMapFromFile(filepath.Join(path, fileName))
+		if err != nil {
+			return nil, err
+		}
+
+		locale := strings.Split(fileName, ".")[0]
+
+		if IsDefaultLocale(fileName) {
+			defaultLocale = locale
```

Comment:
Do we care to log a message if they accidentally declared more than one default locale?

Reply:
Certainly something we can easily implement in this case. Its worth noting we have a [backlog item](https://github.com/Shopify/checkout-web/issues/9515) for more validations around translations and locales - but that is out of scope of this PR.